api/v1/storagecluster_types.go

```diff
@@ -184,11 +184,21 @@ type ManageCephBlockPools struct {
 	ReconcileStrategy    string `json:"reconcileStrategy,omitempty"`
 	DisableStorageClass  bool   `json:"disableStorageClass,omitempty"`
 	DisableSnapshotClass bool   `json:"disableSnapshotClass,omitempty"`
+	// StorageClassName specifies the name of the storage class created for ceph block pools
+	// +kubebuilder:validation:MaxLength=253
+	StorageClassName string `json:"storageClassName,omitempty"`
```

Comment:
We should add more validations to names. Kubernetes has multiple restrictions on naming. We should follow that to not hit into unwanted issue.
Ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

Reply:
In that case, we would have to use a pattern validation like this
```
// StorageClassName specifies the name of the storage class created for cephfs
// +kubebuilder:validation:MaxLength=253
// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
StorageClassName string `json:"storageClassName,omitempty"`
```
Ref: https://github.com/kubernetes/apimachinery/blob/v0.27.3/pkg/util/validation/validation.go#L215

Does this look fine?

Reply:
Or, should we just use the `validation` pkg from `apimachinery` and validate the name in the `validateStorageClassname` function?

Reply:
Pattern validation is good enough.